### DISCUSSION TITLE
Is Pieces App on Linux a Linux desktop environment only app?

### DISCUSSION THREAD

**Discussion by Seefer** [pieces-app/support#95] [Q&A] [answered]
I run VS Code with the Pieces extension in a Ubuntu 22.04.3 distro using Microsoft's WSL. When I try to access either the Pieces Explorer or the Pieces Copilot extensions, the extension windows just seem to show an animated progress indicator, presumably relating to connection?

I followed the instructions for installing Pieces OS on Linux and I do eventually see a time-out VS Code pup-up showing it failed to connect to pieces OS and showing a Launch Pieces OS button but clicking on that does not solve the issue.

If I run `pieces-os` on my bash command-line I get the following error:

`2024/02/19 14:49:40.168982 cmd_run.go:1055: WARNING: cannot start document portal: dial unix /run/user/1000/bus: connect: no such file or directory`
 
Is Pieces a Linux desktop environment only tool?

![image](https://github.com/pieces-app/support/assets/129484/6fb82cdb-c3ed-4887-9d66-77db55358a14)
![image](https://github.com/pieces-app/support/assets/129484/8a532746-1aa6-4686-8432-6555a362efb6)
![image](https://github.com/pieces-app/support/assets/129484/2b8296d7-c973-453c-98ff-cda1e2f0ef4f)
![image](https://github.com/pieces-app/support/assets/129484/3ba7772c-1e4b-48dc-9816-5c131018b4d6)

**Comment by caleb-at-pieces** [comment, marked as answer]
Yes, Pieces OS is not capable of running in a headless environment i.e a WSL container.

With that said, the vs code extension is set up to work for the case that you have Pieces OS running on your windows machine, and you are using the WSL feature in vs code!
